packages/vite/src/node/optimizer/index.ts

```diff
@@ -240,48 +247,51 @@ export interface DepOptimizationMetadata {
  * Scan and optimize dependencies within a project.
  * Used by Vite CLI when running `vite optimize`.
  */
+// TODO: do we need this? It is exposed for the CLI command `vite optimize`
```

Comment:
I also think the same. It currently only runs the client optimizer so this command was already outdated after we started allowing optimization of deps for the SSR environment. It is currently working though, so maybe we can remove it after Vite 6 to avoid a potential breaking change. It seems there is [some usage](https://github.com/search?q=%22vite+optimize%22&type=code) out there. I don't know why people are doing this though, I imagine if we make this command a noop, all these repos will keep working.

For now, I removed these TODOs. We can revisit this before the Vite 6 though. Making it a noop with a message that says this command is a noop would be ok to me.